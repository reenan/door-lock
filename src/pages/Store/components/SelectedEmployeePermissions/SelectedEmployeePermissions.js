import React, { Component } from 'react'
import PropTypes from 'prop-types'

class SelectedEmployeePermissions extends Component {
  render() {
    const { selectedEmployee, roles, doors, doorsKeys } = this.props

    return (
      <div className='selected-employee-permissions'>
        {
          doorsKeys.map((doorID) => (
            <div key={doorID} className='door-permissions'>
              {
                roles[selectedEmployee.role].permissions[doorID] ?
                <span className='warn green'>Allowed to open: </span> :
                <span className='warn red'>Not allowed to open: </span> 
              }
              <span>{doors[doorID].name}</span>
            </div>
          ))
        }
      </div>
    )
  }
}

SelectedEmployeePermissions.propTypes = {
  selectedEmployee: PropTypes.object.isRequired,
  doorsKeys: PropTypes.array.isRequired,
  doors: PropTypes.object.isRequired,
  roles: PropTypes.object.isRequired,
}

export default SelectedEmployeePermissions