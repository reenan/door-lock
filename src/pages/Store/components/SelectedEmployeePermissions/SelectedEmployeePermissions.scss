.selected-employee-permissions {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(250px, 1fr));
  grid-gap: 15px;

  .door-permissions {
    line-height: 16px;
    border: 1px solid #fff;
    padding: 15px;
    border-radius: 5px;
    background-color: #0000001a;

    span:not(.warn) {
      margin-left: 5px;
      font-size: 15px;
    }
  }
}
