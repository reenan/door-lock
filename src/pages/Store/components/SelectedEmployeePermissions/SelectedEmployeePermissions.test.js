import React from 'react'
import SelectedEmployeePermissions from './SelectedEmployeePermissions'

import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const component = renderer.create(
    <SelectedEmployeePermissions
      selectedEmployee={{}}
      doorsKeys={[]}
      doors={{}}
      roles={{}}
    />
  )

  expect(component).toMatchSnapshot()
})

it('renders without crashing with values', () => {
  const component = renderer.create(
    <SelectedEmployeePermissions
      selectedEmployee={{ role: 'mock' }}
      doorsKeys={['mock']}
      doors={{ mock: { name: 'Mock' } }}
      roles={{ mock: { permissions: { mock: false } } }}
    />
  )

  expect(component).toMatchSnapshot()
})

it('renders without crashing with values and permission', () => {
  const component = renderer.create(
    <SelectedEmployeePermissions
      selectedEmployee={{ role: 'mock' }}
      doorsKeys={['mock']}
      doors={{ mock: { name: 'Mock' } }}
      roles={{ mock: { permissions: { mock: true } } }}
    />
  )

  expect(component).toMatchSnapshot()
})
