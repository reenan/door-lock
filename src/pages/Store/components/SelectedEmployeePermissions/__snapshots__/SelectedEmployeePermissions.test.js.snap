// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="selected-employee-permissions"
/>
`;

exports[`renders without crashing with values 1`] = `
<div
  className="selected-employee-permissions"
>
  <div
    className="door-permissions"
  >
    <span
      className="warn red"
    >
      Not allowed to open: 
    </span>
    <span>
      Mock
    </span>
  </div>
</div>
`;

exports[`renders without crashing with values and permission 1`] = `
<div
  className="selected-employee-permissions"
>
  <div
    className="door-permissions"
  >
    <span
      className="warn green"
    >
      Allowed to open: 
    </span>
    <span>
      Mock
    </span>
  </div>
</div>
`;
